web_src/js/components/RepoActionView.vue

```diff
@@ -235,32 +241,46 @@ const sfc = {
           ]);
         } catch (err) {
           if (err instanceof TypeError) return; // avoid network error while unloading page
-          throw err;
+          showErrorToast(err.message, {preventDuplicates: true});
+          // reset all step loading states, we can't easily tell which one failed at this point
+          for (let i = 0; i < this.currentJob.steps.length; i++) {
+            if (this.currentJobStepsStates[i].loading) {
+              this.currentJobStepsStates[i].loading = false;
+            }
+          }
         }
 
-        this.artifacts = artifacts['artifacts'] || [];
-
-        // save the state to Vue data, then the UI will be updated
-        this.run = job.state.run;
-        this.currentJob = job.state.currentJob;
+        if (artifacts) {
+          this.artifacts = artifacts['artifacts'] || [];
+        }
 
-        // sync the currentJobStepsStates to store the job step states
-        for (let i = 0; i < this.currentJob.steps.length; i++) {
-          if (!this.currentJobStepsStates[i]) {
-            // initial states for job steps
-            this.currentJobStepsStates[i] = {cursor: null, expanded: false};
+        if (job) {
```

Comment:
Here and above: why it needs new two `if` here? They are from `[job, artifacts] = await Promise.all`, and the response shouldn't be null.

Reply:
If the `await Promise.all` fails, none of these 2 variables could have value.

So it should return early in the `catch` block above, but not use extra `if` blocks here.

Reply:
I think the root problem is that the state management in this `loadData/loadJob` is quite messy (because after the initial version, then people only add code to "patch" it).

As a bug fix, I think this PR could work. While if we'd like to make the code right, I guess we need to rewrite the code and re-design the state management logic. For example, maybe `this.loading` doesn't make sense now, we need to track every request's state separately.

(Just some new thoughts)

Reply:
The `if`s are meant to perform a partial state update if one of the requests fails.

Reply:
But I see you are right, `Promise.all` can not deliver partial results, it either resolves both or rejects with the first encountered error, so my interpretion of the return value was wrong and the `if`s are not needed.